### ISSUE TITLE
AG WG Review: 2.4.13 Focus Appearance (Level AAA) WCAG2ICT Guidance

### ISSUE THREAD

**Issue by pday1** [w3c/wcag2ict#978]
## Summary
Review of proposed WCAG2ICT guidance for the Level AAA WCAG SC: 2.4.13 Focus Appearance, as incorporated into the Editor's draft. 

## Maturity Level
Refining

## Action Needed
* Review content (See Content for Review section below for details.)
* Use the thumbs up and thumbs down icons to indicate agreement or disagreement with various comments. Do not add comments to simply support other comments. 
* Note any suggested edits for improvement and/or reasoning either in a comment on this issue or by creating a PR on the WCAG2ICT document.  
* Level A and AA SCs can be found in the [comments-by-guideline-and-success-criteria.md](https://github.com/w3c/wcag2ict/blob/main/comments-by-guideline-and-success-criterion.md) file.
* Level AAA SCs can be found in the [comments-on-level-AAA-success-criteria.md](https://github.com/w3c/wcag2ict/blob/main/comments-on-level-AAA-success-criteria.md)
----

## Background information 

Once finalized, this new WCAG2ICT Note will build on the previous version by also covering WCAG 2.2 level AAA success criteria (currently only level A and AA criteria are covered.  This is important as different jurisdictions are using different versions of WCAG for the accessibility of information and communication technology. 

## Content for Review

* Section [Applying 2.4.13 Focus Appearance to non-web documents and non-web software](https://w3c.github.io/wcag2ict/#applying-2-4-13-focus-appearance-to-non-web-documents-and-non-web-software).  

## Reference Material
In case you need them, links to various resources:

* [2.4.13 Focus Appearance](https://www.w3.org/TR/WCAG22/#focus-appearance) SC text in WCAG 2.2
* [Intent section of Understanding 2.4.13 Focus Appearance](https://www.w3.org/WAI/WCAG22/Understanding/focus-appearance.html#intent)

**Comment by maryjom** [comment]
Editorial comments: 
- Replacement words "user agent or platform software" are inconsistent with other SCs in WCAG2ICT which say "user agent or other platform software".  This was due to feedback from prior to the previous publication because browsers are given as an example of a software platform. 
- Notes 1, 2, and 3 indicate that they were "ADDED" by WAG2ICT, but these notes actually originate from WCAG. Remove the (ADDED) notation

Substantive comment:
Similar to SC 2.4.12 guidance, Note 5 seems to be copied without modification to apply to Focus Appearance. It also talks about Keyboard Trap which is unrelated to this SC.

**Comment by pday1** [comment]
Current content in editor's draft:

> [Applying 2.4.13 Focus Appearance to non-web documents and non-web software](https://w3c.github.io/wcag2ict/#applying-2-4-13-focus-appearance-to-non-web-documents-and-non-web-software)
> 
> This applies directly as written, and as described in [Intent from Understanding Success Criterion 2.4.13](https://www.w3.org/WAI/WCAG22/Understanding/focus-appearance.html#intent), replacing “user agent” with “user agent or platform software”.
> 
> With these substitutions, it would read:
> 
> 2.4.13 Focus Appearance: When the keyboard [focus indicator](https://www.w3.org/TR/WCAG22/#dfn-focus-indicator) is visible, an area of the focus indicator meets all the following:
> 
> is at least as large as the area of a 2 [CSS pixel](https://www.w3.org/TR/WCAG22/#dfn-css-pixels) thick [perimeter](https://www.w3.org/TR/WCAG22/#dfn-perimeter) of the unfocused component or sub-component, and
> has a contrast ratio of at least 3:1 between the same pixels in the focused and unfocused states.
> Exceptions:
> 
> The focus indicator is determined by the [[user agent](https://www.w3.org/TR/WCAG22/#dfn-user-agents) or platform software] and cannot be adjusted by the author, or
> The focus indicator and the indicator's background color are not modified by the author.
> NOTE 1 (ADDED)
> What is perceived as the user interface component or sub-component (to determine enclosure or size) depends on its visual [presentation](https://www.w3.org/TR/WCAG22/#dfn-presentation). The visual presentation includes the component's visible [content](https://www.w3.org/TR/WCAG22/#dfn-content), border, and component-specific background. It does not include shadow and glow effects outside the component's content, background, or border.
> 
> NOTE 2 (ADDED)
> Examples of sub-components that may receive a focus indicator are menu items in an opened drop-down menu, or focusable cells in a grid.
> NOTE 3 (ADDED)
> Contrast calculations can be based on colors defined within the [technology](https://www.w3.org/TR/WCAG22/#dfn-technologies) (such as HTML, CSS and SVG). Pixels modified by [user agent or platform software] resolution enhancements and anti-aliasing can be ignored.
> 
> NOTE 4 (ADDED)
> In technologies where CSS is not used, the definition of 'CSS pixel' applies as described in [Applying “CSS pixel” to non-web documents and non-web software](https://w3c.github.io/wcag2ict/#applying-css-pixel-to-non-web-documents-and-non-web-software).
> 
> NOTE 5 (ADDED) (FOR NON-WEB SOFTWARE)
> This criterion applies when focus can be moved using a keyboard interface. Some software may accept input from a keyboard, keypad, or controller, yet not offer any mechanism for focus; for example, the keys are mapped directly to functions without moving focus between on-screen controls. In this case, there is no concept of focus, and therefore keyboard traps cannot exist and this success criterion would be satisfied.
> NOTE 6 (ADDED) (FOR NON-WEB SOFTWARE)
> For general guidance, see also the [comments on closed functionality](https://w3c.github.io/wcag2ict/#comments-on-closed-functionality). For commentary on the individual success criterion, see [Success criteria problematic for closed functionality, Focus Appearance](https://w3c.github.io/wcag2ict/#problematic-for-closed-focus-appearance).

**Comment by pday1** [comment]
Proposed content (changes in **bold**):

> [Applying 2.4.13 Focus Appearance to non-web documents and non-web software](https://w3c.github.io/wcag2ict/#applying-2-4-13-focus-appearance-to-non-web-documents-and-non-web-software)
> 
> This applies directly as written, and as described in [Intent from Understanding Success Criterion 2.4.13](https://www.w3.org/WAI/WCAG22/Understanding/focus-appearance.html#intent), replacing “user agent” with “user agent or **other** platform software”.
> 
> With these substitutions, it would read:
> 
> 2.4.13 Focus Appearance: When the keyboard [focus indicator](https://www.w3.org/TR/WCAG22/#dfn-focus-indicator) is visible, an area of the focus indicator meets all the following:
> 
> is at least as large as the area of a 2 [CSS pixel](https://www.w3.org/TR/WCAG22/#dfn-css-pixels) thick [perimeter](https://www.w3.org/TR/WCAG22/#dfn-perimeter) of the unfocused component or sub-component, and
> has a contrast ratio of at least 3:1 between the same pixels in the focused and unfocused states.
> Exceptions:
> 
> The focus indicator is determined by the [[user agent](https://www.w3.org/TR/WCAG22/#dfn-user-agents) or **other** platform software] and cannot be adjusted by the author, or
> The focus indicator and the indicator's background color are not modified by the author.
> NOTE 1 ~~(ADDED)~~
> What is perceived as the user interface component or sub-component (to determine enclosure or size) depends on its visual [presentation](https://www.w3.org/TR/WCAG22/#dfn-presentation). The visual presentation includes the component's visible [content](https://www.w3.org/TR/WCAG22/#dfn-content), border, and component-specific background. It does not include shadow and glow effects outside the component's content, background, or border.
> 
> NOTE 2 ~~(ADDED)~~
> Examples of sub-components that may receive a focus indicator are menu items in an opened drop-down menu, or focusable cells in a grid.
> NOTE 3 ~~(ADDED)~~
> Contrast calculations can be based on colors defined within the [technology](https://www.w3.org/TR/WCAG22/#dfn-technologies) (such as HTML, CSS and SVG). Pixels modified by [user agent or platform software] resolution enhancements and anti-aliasing can be ignored.
> 
> NOTE 4 (ADDED)
> In technologies where CSS is not used, the definition of 'CSS pixel' applies as described in [Applying “CSS pixel” to non-web documents and non-web software](https://w3c.github.io/wcag2ict/#applying-css-pixel-to-non-web-documents-and-non-web-software).
> 
> ~~NOTE 5 (ADDED) (FOR NON-WEB SOFTWARE)
> This criterion applies when focus can be moved using a keyboard interface. Some software may accept input from a keyboard, keypad, or controller, yet not offer any mechanism for focus; for example, the keys are mapped directly to functions without moving focus between on-screen controls. In this case, there is no concept of focus, and therefore keyboard traps cannot exist and this success criterion would be satisfied.~~
> NOTE 6 (ADDED) (FOR NON-WEB SOFTWARE)
> For general guidance, see also the [comments on closed functionality](https://w3c.github.io/wcag2ict/#comments-on-closed-functionality). For commentary on the individual success criterion, see [Success criteria problematic for closed functionality, Focus Appearance](https://w3c.github.io/wcag2ict/#problematic-for-closed-focus-appearance).
